src/catalog/database_catalog.cpp

```diff
@@ -20,6 +24,174 @@
+    return false;  // invalid object
+  }
+  // std::lock_guard<std::mutex> lock(table_cache_lock);
+
+  // check if already in cache
+  if (table_objects_cache.find(table_object->table_oid) !=
+      table_objects_cache.end()) {
+    LOG_DEBUG("Table %u already exists in cache!", table_object->table_oid);
+    return false;
+  }
+
+  if (table_name_cache.find(table_object->table_name) !=
+      table_name_cache.end()) {
+    LOG_DEBUG("Table %s already exists in cache!",
+              table_object->table_name.c_str());
+    return false;
+  }
+
+  table_objects_cache.insert(
+      std::make_pair(table_object->table_oid, table_object));
+  table_name_cache.insert(
+      std::make_pair(table_object->table_name, table_object));
+  return true;
+}
+
+/* @brief   evict table catalog object from cache
+ * @param   table_oid
+ * @return  true if table_oid is found and evicted; false if not found
+ */
+bool DatabaseCatalogObject::EvictTableObject(oid_t table_oid) {
+  // std::lock_guard<std::mutex> lock(table_cache_lock);
+
+  // find table name from table name cache
+  auto it = table_objects_cache.find(table_oid);
+  if (it == table_objects_cache.end()) {
+    return false;  // table oid not found in cache
+  }
+
+  auto table_object = it->second;
+  PL_ASSERT(table_object);
+  table_objects_cache.erase(it);
+  table_name_cache.erase(table_object->table_name);
+  return true;
+}
+
+/* @brief   evict table catalog object from cache
+ * @param   table_name
+ * @return  true if table_name is found and evicted; false if not found
+ */
+bool DatabaseCatalogObject::EvictTableObject(const std::string &table_name) {
+  // std::lock_guard<std::mutex> lock(table_cache_lock);
+
+  // find table name from table name cache
+  auto it = table_name_cache.find(table_name);
+  if (it == table_name_cache.end()) {
+    return false;  // table oid not found in cache
+  }
+
+  auto table_object = it->second;
+  PL_ASSERT(table_object);
+  table_name_cache.erase(it);
+  table_objects_cache.erase(table_object->table_oid);
+  return true;
+}
+
+/*@brief   evict all table catalog objects in this database from cache
+*/
+void DatabaseCatalogObject::EvictAllTableObjects() {
+  // std::lock_guard<std::mutex> lock(table_cache_lock);
+  table_objects_cache.clear();
+  table_name_cache.clear();
+}
+
+/* @brief   Get table catalog object from cache or all the way from storage
+ * @param   table_oid     table oid of the requested table catalog object
+ * @param   cached_only   if cached only, return nullptr on a cache miss
+ * @return  Shared pointer to the requested table catalog object
+ */
+std::shared_ptr<TableCatalogObject> DatabaseCatalogObject::GetTableObject(
+    oid_t table_oid, bool cached_only) {
+  auto it = table_objects_cache.find(table_oid);
+  if (it != table_objects_cache.end()) return it->second;
+
+  if (cached_only) {
+    // cache miss return empty object
+    return nullptr;
+  } else {
+    // cache miss get from pg_table
```

Comment:
The comment in if and else can be removed and added in @brief section of comment block above.